src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -225,28 +234,48 @@ public synchronized AmplitudeClient initialize(final Context context, final Stri
         runOnLogThread(new Runnable() {
             @Override
             public void run() {
-                if (!client.initialized) {
-                    AmplitudeClient.upgradePrefs(context);
-                    AmplitudeClient.upgradeSharedPrefsToDB(context);
-                    client.httpClient = new OkHttpClient();
-                    client.initializeDeviceInfo();
-
-                    if (userId != null) {
-                        client.userId = userId;
-                        client.dbHelper.insertOrReplaceKeyValue(USER_ID_KEY, userId);
-                    } else {
-                        client.userId = client.dbHelper.getValue(USER_ID_KEY);
-                    }
-                    Long optOut = client.dbHelper.getLongValue(OPT_OUT_KEY);
-                    client.optOut = optOut != null && optOut == 1;
+                if (!initialized) {
+                    try {
+                        AmplitudeClient.upgradePrefs(context);
```

Comment:
verify that all steps in this try/catch block are idempotent and add a comment that they should be